pkg/collector/kube.go

```diff
@@ -21,30 +21,48 @@ func newKubeCollector(kubeconfig string, kubecontext string, discoveryClient dis
 	if discoveryClient != nil {
 		col.discoveryClient = discoveryClient
 	} else {
-		pathOptions := clientcmd.NewDefaultPathOptions()
-		if kubeconfig != "" {
-			pathOptions.GlobalFile = kubeconfig
+		var err error
```

Comment:
is this var needed?

Reply:
Kind of yes, as it is, if I don't declare the variable, the `if` below would fail on undefined var. The issue is the choice of operator `:=` vs `=`, and so both of the variables on the left side have to be either undefined or exist. I guess I can refactor both to:
```go
if restConfig, err := newClientRestConfig(kubeconfig, kubecontext, rest.InClusterConfig); err != nil {
    return nil, fmt.Errorf("failed to assemble client config: %w", err)
}
col.restConfig = restconfig
```

In the end it will result in extra line, as it would have to be done for both `if`s, but perhaps it's nicer? WDYT?

Reply:
I prefer not having a var, and using:

```
:=
```

Reply:
For the record settled on keeping as is - my example above wouldn't work, and with `:=` one would have to drop the short form, as otherwise the `restConfig` would be in local if scope only.